### PR TITLE
Turn rules into npm package

### PR DESCRIPTION
**Pull request by Ellpeck**
Followed this lovely tutorial [here](https://www.freecodecamp.org/news/creating-your-own-eslint-config-package/), for future reference

### CODE REVIEW

**Comment by Ellpeck** [comment]
actually, i want to reuse this pull request to add ci workflows too!

**Comment by Ellpeck** [comment]
For this, we need a `GITHUB_PUSH_TOKEN` for pushing the patch to main, and a `NPM_TOKEN` for the NPM registry.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  push:
+    branches: [main]
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_PUSH_TOKEN }}
```
Comment (EagleoutIce) [inline .github/workflows/release.yml]:
```suggestion
          token: ${{ secrets.PUSH_TOKEN }}
```

**Review by EagleoutIce — CHANGES_REQUESTED** [review]
Let's try

**Pull request merged**